server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -738,7 +731,7 @@ protected void writeIndexGen(final RepositoryData repositoryData, final long rep
         // write the index file
         final String indexBlob = INDEX_FILE_PREFIX + Long.toString(newGen);
         logger.debug("Repository [{}] writing new index generational blob [{}]", metadata.name(), indexBlob);
-        writeAtomic(indexBlob, snapshotsBytes, true);
+        writeAtomic(indexBlob, snapshotsBytes, false);
```

Comment:
This should be idempotent anyway, all we're doing here is potentially preventing deleting the old index file or updating the latest blob file in subsequent steps if either of those failed before a master failover.